sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,171 @@
+    'DPGMM',
+    'GMM',
+    'log_multivariate_normal_density',
+    'sample_gaussian',
+]
+
+_TAB_IGNORES = [
+]
+
+# Methods to test for, in any class
+_METHODS_IGNORE_NONE_Y = [
+        'fit',
+        'score',
+        'fit_predict',
+        'fit_transform',
+        'partial_fit',
+        'predict'
+]
+
+
+def test_docstring_parameters():
+    """Test module docstring formatting."""
+    try:
+        import numpydoc  # noqa
+    except ImportError:
+        raise SkipTest(
+            "numpydoc is required to test the docstrings")
+
+    from numpydoc import docscrape
+
+    incorrect = []
+    for name in PUBLIC_MODULES:
+        with warnings.catch_warnings(record=True):  # traits warnings
+            module = __import__(name, globals())
+        for submod in name.split('.')[1:]:
+            module = getattr(module, submod)
+        classes = inspect.getmembers(module, inspect.isclass)
+        for cname, cls in classes:
+            this_incorrect = []
+            if cname in _DOCSTRING_IGNORES:
+                continue
+            if cname.startswith('_') and cname not in _DOC_SPECIAL_MEMBERS:
+                continue
+            with warnings.catch_warnings(record=True) as w:
+                cdoc = docscrape.ClassDoc(cls)
+            if len(w):
+                raise RuntimeError('Error for __init__ of %s in %s:\n%s'
+                                   % (cls, name, w[0]))
+            if hasattr(cls, '__init__'):
+                this_incorrect += check_parameters_match(cls.__init__, cdoc,
+                                                         class_name=cname)
+
+            for method_name in cdoc.methods:
+                method = getattr(cls, method_name)
+                param_ignore = None
+                # Now skip docstring test for y when y is None
+                # by default for API reason
+                if method_name in _METHODS_IGNORE_NONE_Y:
+                    sig = signature(method)
+                    if ('y' in sig.parameters and
+                            sig.parameters['y'].default is None):
+                        param_ignore = ['y']  # ignore y for fit and score
+                result = check_parameters_match(method, ignore=param_ignore,
+                                                class_name=cname)
+                this_incorrect += result
+
+            if hasattr(cls, '__call__'):
+                this_incorrect += check_parameters_match(cls.__call__,
+                                                         class_name=cname)
+
+            incorrect += this_incorrect
+
+        functions = inspect.getmembers(module, inspect.isfunction)
+        for fname, func in functions:
+            # Don't test private methods / functions
+            if fname.startswith('_'):
+                continue
+            name_ = get_func_name(func)
+            if not any(d in name_ for d in _DOCSTRING_IGNORES) and \
+                    'deprecation_wrapped' not in func.__code__.co_name:
```

Comment:
I've modified this into a helper which checks for 'deprecated' string in `func.__closure__` if `__closure__` attribute exists for that function. Let me know if that helper looks good.